src/App.vue

```diff
@@ -90,26 +111,126 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
+  private nameRequestInvalidType: string = ''
+  private evaluatingPreconditions: boolean = true
 
   private async created (): Promise<any> {
-    // Mock the nrNumber and Data:
-    this.setNameRequestState({ nrNumber: 'NR7654560', entityType: 'BC' })
+    // Mock the nrNumber and Data
+    const nameRequest = await this.evaluateNRPreconditions()
+    if (nameRequest && nameRequest.nrNum && nameRequest.isConsumable) {
+      // TODO: Handling different NR Formats
+      // Will probably change once we proxy through legal API with a consistent format
+      this.setNameRequestState({ nrNumber: nameRequest.nrNum.replace('NR ', ''), entityType: 'BC', filingId: null })
+      this.setCurrentDate(this.dateToUsableString(new Date()))
 
-    this.setCurrentDate(this.dateToUsableString(new Date()))
+      try {
+        // Retrieve draft filing if it exists for the nrNumber specified
+        this.draftFiling = await this.fetchDraft()
 
-    try {
-      // Retrieve draft filing if it exists for the nrNumber specified
-      this.draftFiling = await this.fetchDraft()
+        // Parse the draft data into the store if it exists
+        this.draftFiling && this.parseDraft(this.draftFiling)
 
-      // Parse the draft data into the store if it exists
-      this.draftFiling && this.parseDraft(this.draftFiling)
-    } catch (e) {
-      // TODO: Catch a flag from the api, if there is an error to be handled.
+        // Inform the router view we are resuming a draft and to update ui
+        this.isDraft = true
+      } catch (e) {
+        // TODO: Catch a flag from the api, if there is an error to be handled.
+      }
     }
+    this.evaluatingPreconditions = false
+  }
+
+  /**
+   * Redirect to business URL
+   */
+  private redirectToBusinessUrl () : void {
+    const businessUrl: string = sessionStorage.getItem('BUSINESSES_URL') || ''
+    window.location.assign(businessUrl)
+  }
+
+  private authAPIURL () {
+    return sessionStorage.getItem('AUTH_API_URL')
+  }
+  /**
+   * Fetch Authorizations by NR number
+   */
+  private getNRAuthorizations (nrNumber) {
+    const url = nrNumber + '/authorizations'
+    const config = {
+      baseURL: this.authAPIURL() + 'entities/'
+    }
+    return axios.get(url, config)
+  }
+
+  private storeAuthRoles (response) {
+    // NB: roles array may contain 'view', 'edit' or nothing
+    const authRoles = response && response.data && response.data.roles
+    if (authRoles && authRoles.length > 0) {
+      this.setAuthRoles(authRoles)
+    } else {
+      throw new Error('Invalid auth roles')
+    }
+  }
+
+  /**
+   * Evaluate Name Request pre conditions
+   */
+  private async evaluateNRPreconditions () {
+    // Clear error conditions in the event this is invoked more than one time (retry)
+    this.nameRequestInvalidErrorDialog = false
+    this.accountAuthorizationErrorDialog = false
+
+    const queryNrNumber : string = this.$route.query.nrNumber as string
```

Comment:
"as" may be redundant. I think (but not certain) that route query and and params are strings.